pkg/types/baremetal/platform.go

```diff
@@ -21,6 +31,7 @@ type Host struct {
 	BootMACAddress  string                    `json:"bootMACAddress" validate:"required,uniqueField"`
 	HardwareProfile string                    `json:"hardwareProfile"`
 	RootDeviceHints *v1alpha1.RootDeviceHints `json:"rootDeviceHints,omitempty"`
+	BootMode        BootMode                  `json:"bootMode,omitempty"`
```

Comment:
Would love to see static validations on allowed enum values

Reply:
I'm not sure what that means. Is there an example of another field where that's being done?

Reply:
Something like https://github.com/openshift/installer/blob/0224256a000774650771e00a447dad932a891d42/pkg/types/validation/machinepools.go#L64

Reply:
Oh, sure, I thought "static validations" was some sort of golang thing I hadn't encountered yet. :-)

I added a patch to the PR to validate the boot mode when provisioning the cluster. @stbenjam @kirankt does that seem like the right time, or should it come during `ValidatePlatform`?

Reply:
@dhellmann IMHO think this should belong in ValidatePlatform just because we defer pretty much everything related to provisoning to ValidateProvisoning and this field is not related to it.

Reply:
Is boot mode used when we don't provision?

Reply:
> Is boot mode used when we don't provision?

No, this should only be validated when we're actually going to provision?

Reply:
I went ahead and moved the validation to the earlier phase. It's not expense and not wrong to do it there. The assisted installer probably won't set the boot mode at all, but that doesn't really matter.